client/src/main/java/io/pravega/client/tables/KeyValueTable.java

```diff
@@ -0,0 +1,247 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.client.tables;
+
+import io.pravega.common.util.AsyncIterator;
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Defines all operations that are supported on a Key-Value Table.
+ * <p>
+ * A Key-Value Table is a distributed Key-Value Store that indexes Entries by Keys. It uses Table Segments (non-distributed
+ * Key-Value Store backed by a single Pravega Segment) as the fundamental storage primitive and provides a unified view
+ * of all Table Segments involved. Each {@link TableKey} is hashed to a Table Partition which may be represented by one
+ * or more Table Segments (depending on the Key-Value Table configuration chosen when it was created). Such partitioning
+ * enables the Key-Value Table to be distributed across the Pravega cluster but also introduces some constraints for
+ * certain operations (such as multi-key/entry atomic updates). See below for details.
+ * <p>
+ * Key Families are used to group related Keys together in the same Table Partition, which allows multiple
+ * keys/entries belonging to the same Key Family to be updated/removed atomically.
+ * <ul>
+ * <li> Multiple Keys/Entries in the same Key Family can be updated or removed atomically (either all at once or none).
+ * <li> Iterating through all Keys/Entries in the same Key Family is possible.
+ * <li> The same Key may exist in multiple Key Families or even not be associated with any Key Family at all. Such keys
+ * are treated as distinct keys and will not interfere with each other (i.e., if key K1 exists in Key Families F1 and F2,
+ * then F1.K1 is different from F2.K1 and both are different from K1 (no Key Family association).
+ * <li> Keys that do not belong to any Key Family will be uniformly distributed across the Key-Value Table Partitions and
+ * cannot be used for multi-key/entry atomic updates or removals or be iterated on.
+ * <li> Improper use of Key Families may result in degraded performance. If a disproportionate number of Keys/Entries
```

Comment:
This is explaining that "improper" use is bad, but it does so without first explaining how the system is architected and what a 'proper' use looks like given that architecture.

Reply:
It is explained a few lines above:

```
 Each {@link TableKey} is hashed to a Table Partition which may be represented by one or more Table Segments

...

Key Families are used to group related Keys together in the same Table Partition, which allows multiple keys/entries belonging to the same Key Family to be updated/removed atomically.
```

Reply:
Ok. I rephrased the entire paragraph.